### PR TITLE
Add SuspenseList to react-is

### LINKED ISSUE #20868: react-is lacks SuspenseList
I'd like to be able to detect `SuspenseList` using `react-is`. Is there any reason why I shouldn't?

### PR DESCRIPTION
**Pull request by bvaughn**
Resolves #20868

### CODE REVIEW

**Comment by eps1lon** [comment]
`react` has different entrypoints (`index.experimental` and `index.stable`) which probably makes more sense to use here as well since then the stable OSS builds won't list these experimental features at all.

**Comment by bvaughn** [comment]
Not sure I understand.

Those entry points (forks) are what's used by our Rollup build script to generate different NPM bundles, but we don't reference them in tests. We use `@gate` pragma to enable/disable tests.

Hm. Not sure how to get it working right in this case. I thought I had it but...

**Comment by eps1lon** [comment]
The idea was that you don't branch inside of `isSuspenseList` and `SuspenseList = ...` but instead have `index.js` exporting everything from `ReactIS`, `index.experimental` exporting everything as well, and `index.stable` (assuming this is the OSS build) exporting everything but the `SuspenseList` related stuff. 

<details>
<summary>patch</summary>

```diff
diff --git a/packages/react-is/index.experimental.js b/packages/react-is/index.experimental.js
new file mode 100644
index 000000000..e9934e91f
--- /dev/null
+++ b/packages/react-is/index.experimental.js
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+'use strict';
+
+export {
+  isAsyncMode,
+  isConcurrentMode,
+  ContextConsumer,
+  isContextConsumer,
+  ContextProvider,
+  isContextProvider,
+  Element,
+  isElement,
+  ForwardRef,
+  isForwardRef,
+  Fragment,
+  isFragment,
+  Lazy,
+  isLazy,
+  Memo,
+  isMemo,
+  Portal,
+  isPortal,
+  Profiler,
+  isProfiler,
+  StrictMode,
+  isStrictMode,
+  Suspense,
+  isSuspense,
+  isValidElementType,
+  // exposeConcurrentModeAPIs
+  SuspenseList,
+  isSuspenseList,
+} from './src/ReactIs';
diff --git a/packages/react-is/index.stable.js b/packages/react-is/index.stable.js
new file mode 100644
index 000000000..efaed673c
--- /dev/null
+++ b/packages/react-is/index.stable.js
@@ -0,0 +1,38 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+'use strict';
+
+export {
+  isAsyncMode,
+  isConcurrentMode,
+  ContextConsumer,
+  isContextConsumer,
+  ContextProvider,
+  isContextProvider,
+  Element,
+  isElement,
+  ForwardRef,
+  isForwardRef,
+  Fragment,
+  isFragment,
+  Lazy,
+  isLazy,
+  Memo,
+  isMemo,
+  Portal,
+  isPortal,
+  Profiler,
+  isProfiler,
+  StrictMode,
+  isStrictMode,
+  Suspense,
+  isSuspense,
+  isValidElementType,
+} from './src/ReactIs';
diff --git a/packages/react-is/src/ReactIs.js b/packages/react-is/src/ReactIs.js
index 9aa1aa23d..dd81ec036 100644
--- a/packages/react-is/src/ReactIs.js
+++ b/packages/react-is/src/ReactIs.js
@@ -72,9 +72,7 @@ export const Portal = REACT_PORTAL_TYPE;
 export const Profiler = REACT_PROFILER_TYPE;
 export const StrictMode = REACT_STRICT_MODE_TYPE;
 export const Suspense = REACT_SUSPENSE_TYPE;
-export const SuspenseList = __EXPERIMENTAL__
-  ? REACT_SUSPENSE_LIST_TYPE
-  : undefined;
+export const SuspenseList = REACT_SUSPENSE_LIST_TYPE;
 
 export {isValidElementType};
 
@@ -146,9 +144,5 @@ export function isSuspense(object: any) {
   return typeOf(object) === REACT_SUSPENSE_TYPE;
 }
 export function isSuspenseList(object: any) {
-  if (__EXPERIMENTAL__) {
-    return typeOf(object) === REACT_SUSPENSE_LIST_TYPE;
-  } else {
-    return false;
-  }
+  return typeOf(object) === REACT_SUSPENSE_LIST_TYPE;
 }

```
</details>

Can't test it right now since I'm on a windows machine and the failing CI tasks crash with `Could not locate module scheduler mapped as: C:\Users\eps1lon\Development\react\build2\oss-experimental\scheduler`

**Comment by bvaughn** [comment]
Ah, sorry. I think I misunderstood your question to be about testing. Sounds like you were just suggesting it may be more appropriate to add a fork for `react-is` to expose the new methods. Yeah, that seems like a reasonable suggestion too.

File: packages/react-is/index.experimental.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  isElement,
+  isForwardRef,
+  isFragment,
+  isLazy,
+  isMemo,
+  isPortal,
+  isProfiler,
+  isStrictMode,
+  isSuspense,
+  isSuspenseList,
```
Comment (rickhanlonii) [inline packages/react-is/index.experimental.js:40]: Should this be `unstable_isSuspenseList` for the feature detection reason?

**Review by rickhanlonii — APPROVED** [review]

File: packages/react-is/index.experimental.js (hunk 1 above)
Comment (bvaughn) [reply packages/react-is/index.experimental.js:40]: Sure.

**Pull request merged**